packages-exp/app-check-exp/src/api.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AppCheck, AppCheckOptions } from './public-types';
+import { ERROR_FACTORY, AppCheckError } from './errors';
+import { getState, setState, AppCheckState } from './state';
+import { FirebaseApp, getApp, _getProvider } from '@firebase/app-exp';
+import { getModularInstance } from '@firebase/util';
+import { AppCheckService } from './factory';
+import { AppCheckProvider } from './types';
+
+declare module '@firebase/component' {
+  interface NameServiceMapping {
+    'app-check-exp': AppCheckService;
+  }
+}
+
+export { ReCaptchaV3Provider, CustomProvider } from './providers';
+
+/**
+ * Activate AppCheck for the given app. Can be called only once per app.
```

Comment:
Is "AppCheck" the product name, or is it App Check?

If we want to refer to an actual literal AppCheck object or entity, then we should use backticks.  However, I think all of the instances in this comment are just references to the product, where we should use the product name.

This would be something to check throughout the PR.

Reply:
Went through manually and got most of them I think.